ambry-store/src/main/java/com/github/ambry/store/StorageManagerMetrics.java

```diff
@@ -97,6 +103,11 @@ public StorageManagerMetrics(MetricRegistry registry) {
     Gauge<Long> compactionsInProgressGauge = compactionsInProgress::longValue;
     registry.register(MetricRegistry.name(CompactionManager.class, "CompactionsInProgress"),
         compactionsInProgressGauge);
+
+    Gauge<Long> storeToSkipGauge =
+        () -> mountPathToStoreSetToSkipInCompactions.values().stream().mapToLong(Set::size).sum();
+    registry.gauge(MetricRegistry.name(CompactionManager.class, "CompactionNumberOfStoresToSkip"),
```

Comment:
I'm not sure how long it will take for the lambda to be calculated, but better to be aware of it in case it delay the metric producing which trigger dotted metric issue?

Reply:
this is fast, it's just a stream operations with no more than 130 elements.